sdk/resourcemanager/Proto.Client/compute/VirtualMachineScaleSetRollingUpgrade.cs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.ResourceManager.Core;
+
+#nullable disable
+
+namespace Proto.Compute
+{
+    /// <summary> A Class representing a VirtualMachineScaleSetRollingUpgrade along with the instance operations that can be performed on it. </summary>
+    public class VirtualMachineScaleSetRollingUpgrade : VirtualMachineScaleSetRollingUpgradeOperations
+    {
+        /// <summary>
+        /// Gets the data representing this VirtualMachine.
+        /// </summary>
+        public RollingUpgradeStatusInfo Data { get; private set; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="VirtualMachineScaleSetRollingUpgrade"/> class.
+        /// </summary>
+        /// <param name="operations"> The client parameters to use in these operations. </param>
+        /// <param name="resource"> The resource that is the target of operations. </param>
+        internal VirtualMachineScaleSetRollingUpgrade(OperationsBase operations, RollingUpgradeStatusInfo resource)
+            : base(operations, resource.Id)
```

Comment:
for consistency with the construction on the VirtualMachineScaleSetOperations class should we pass in operations.Id here?